foreign_cc/make_variant.bzl

```diff
@@ -0,0 +1,35 @@
+"""Helper rules and macros for building projects using the variants of GNU Make, such as MSVC's NMake"""
```

Comment:
This may be a lame reason, but I was thinking it'd be nicer to have this new rule in each module so docs for it are rendered into existing pages as add-ons vs making this it's own isolated page. Unless you feel strongly, could this be moved to `foreign_cc/private/transitions.bzl` and have a macro in `make.bzl`, `configure_make.bzl` and perhaps `cmake.bzl` where you have `${rule}_variant` that calls: 
```python
make_variant(name = name, rule = ${rule}, toolchain = toolchain, **kwargs)
```
?

Reply:
Sure, done